runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Environments.java

```diff
@@ -49,9 +48,7 @@
   private static final EnvironmentIdExtractor DEFAULT_SPEC_EXTRACTOR = (transform) -> null;
 
   private static final String JAVA_SDK_HARNESS_CONTAINER_URL =
-      String.format(
-          "%s-%s",
-          ReleaseInfo.getReleaseInfo().getName(), ReleaseInfo.getReleaseInfo().getVersion());
+      String.format("%s-docker-apache.bintray.io/beam/java", System.getenv("USER"));
```

Comment:
Shouldn't we default to the released container by default?

I could see where you would want to use a PipelineOption to control this.

Reply:
`PipelineOptions` aren't available when we're creating the protos for the transforms.

Reply:
We should really stick with the released container instead of defaulting to something with the users name in it. This Environments class is meant to be shared across runners and specializing for the usage of direct runner doesn't seem to make sense.

Reply:
I've added a comment for why this is the user's container, after talking for a bit with Henning.

This should be the user's container - containers and the SDK should be built from the same source, and using a released container means we don't have any guarantee on how the two will interact.

I don't have a JIRA to link against right now, but adding one would probably be valuable.